README.md

```diff
@@ -78,165 +77,254 @@ And now you should be ready to use every functionality `nvim-treesitter` provide
-- [x] Consistent syntax highlighting (the api is not quite stable yet)
-- [x] Statusline indicator (`require'nvim-treesitter'.statusline(size)`)
+## Refactor: smart rename
 
-You can find the roadmap [here](https://github.com/nvim-treesitter/nvim-treesitter/projects/1).
-The roadmap and all features of this plugin are open to change, and any suggestion will be highly appreciated!
+Renames the symbol under the cursor within the current scope (and current file).
 
-## Available Modules
+```lua
+lua <<EOF
+require'nvim-treesitter.configs'.setup {
+  refactor = {
+    smart_rename = {
+      enable = true,
+      keymaps = {
+        smart_rename = "grr",
+      },
+    },
+  },
+}
+EOF
+```
 
-- `highlight`: Consistent syntax highlighting.
-- `incremental_selection`: Syntax based selection.
-- `refactor.highlight_definitions`: Syntax based definition and usage highlighting.
-- `refactor.smart_rename`: Syntax based definition and usage renaming.
-- `refactor.navigation`: Syntax based definition listing and navigation.
-  * List all definitions
-  * Go to definition
-- `textobjects`: Vim textobjects defined by treesitter queries
+## Refactor: navigation
 
-## Defining Modules
+Provides "go to definition" for the symbol under the cursor,
+and lists the definitions from the current file.
 
-Users and plugin authors can take advantage of modules by creating their own. Modules provide:
+```lua
+lua <<EOF
+require'nvim-treesitter.configs'.setup {
+  refactor = {
+    navigation = {
+      enable = true,
+      keymaps = {
+        goto_definition = "gnd",
+        list_definitions = "gnD",
+      },
+    },
+  },
+}
+EOF
+```
 
-- Treesitter language detection support
-- Attach and detach to buffers
-- Works with all nvim-treesitter commands
+## Text objects: selection
 
-You can use the `define_modules` function to define one or more modules or module groups.
+Define your own text objects mappings
+similar to `ip` (inner paragraph) and `ap` (a paragraph).
 
 ```lua
-require 'nvim-treesitter'.define_modules {
-  my_cool_plugin = {
-    attach = function(bufnr, lang)
-      -- Do cool stuff here
-    end,
-    detach = function(bufnr)
-      -- Undo cool stuff here
-    end,
-    is_supported = function(lang)
-      -- Check if the language is supported
-    end
-  }
+lua <<EOF
+require'nvim-treesitter.configs'.setup {
+  textobjects = {
+    enable = true,
+    keymaps = {
+      -- You can use the capture groups defined in textobjects.scm
+      ["af"] = "@function.outer",
+      ["if"] = "@function.inner",
+      ["ac"] = "@class.outer",
+      ["ic"] = "@class.inner",
+
+      -- Or you can define your own textobjects like this
+      ["iF"] = {
+        python = "(function_definition) @function",
+        cpp = "(function_definition) @function",
+        c = "(function_definition) @function",
+        java = "(method_declaration) @function",
+      },
+    },
+  },
 }
+EOF
 ```
 
-Modules can consist of the following properties:
+## Text objects: swap
```

Comment:
We currently have no plans to create user-visible submodules. We will only split textobjects.lua in different source files. Since everything in this module depends on textobject.scm it makes sense to enable/disable everything related together. The primary reason is however that I don't want to duplicated the lengthy attach/detach method which will be the only thing to remain in textobjects.lua